qt/scientific_interfaces/ISISReflectometry/test/Save/AsciiSaverTest.h

```diff
@@ -0,0 +1,185 @@
+  // This means the constructor isn't called when running other tests
+  static AsciiSaverTest *createSuite() { return new AsciiSaverTest(); }
+  static void destroySuite(AsciiSaverTest *suite) { delete suite; }
+
+  void test_save_ascii_algorithm_called_for_ANSTO_format() {
+    runTestSaveAsciiAlgorithmCalledForFileFormat(NamedFormat::ANSTO);
+  }
+
+  void test_save_ascii_algorithm_called_for_Custom_format() {
+    runTestSaveAsciiAlgorithmCalledForFileFormat(NamedFormat::Custom);
+  }
+
+  void test_save_ascii_algorithm_called_for_ThreeColumn_format() {
+    runTestSaveAsciiAlgorithmCalledForFileFormat(NamedFormat::ThreeColumn);
+  }
+
+  void test_save_ascii_algorithm_called_for_ILLCosmos_format() {
+    runTestSaveAsciiAlgorithmCalledForFileFormat(NamedFormat::ILLCosmos);
+  }
+
+  void test_save_orso_algorithm_called_for_ORSOAscii_format() {
+    runTestSaveORSOAlgorithmCalledForFileFormat(NamedFormat::ORSOAscii);
+  }
+
+  void test_saving_multiple_workspaces_calls_algorithm_once_for_each() {
+    std::vector<std::string> const workspacesToSave = {"ws_1", "ws_2", "ws_3"};
+    runTestSaveAsciiAlgorithmCalledForFileFormat(NamedFormat::ANSTO, workspacesToSave);
+  }
+
+  void test_invalid_save_path_throws_exception() {
+    auto mockSaveAlgorithmRunner = std::make_unique<MockSaveAlgorithmRunner>();
+    auto mockFileHandler = MockFileHandler();
+    auto wsNames = createWorkspaces();
+    std::vector<std::string> logParams;
+    auto formatOptions = createFileFormatOptions(NamedFormat::ANSTO);
+
+    expectIsValidSaveDirectory(mockFileHandler, false);
+
+    auto saver = createSaver(std::move(mockSaveAlgorithmRunner), mockFileHandler);
+    TS_ASSERT_THROWS(saver.save(m_saveDirectory, wsNames, logParams, formatOptions), InvalidSavePath const &);
+  }
+
+private:
+  const bool m_includeHeader = true;
+  const bool m_includeQResolution = false;
+  const std::string m_separator = ",";
+  const std::string m_prefix = "test_";
+  const std::string m_saveDirectory = "Test";
+
+  AsciiSaver createSaver(std::unique_ptr<MockSaveAlgorithmRunner> saveAlgRunner, MockFileHandler &mockFileHandler) {
+    return AsciiSaver(std::move(saveAlgRunner), &mockFileHandler);
+  }
+
+  Workspace2D_sptr createWorkspace(const std::string &name) {
+    Workspace2D_sptr ws = WorkspaceCreationHelper::create2DWorkspace(10, 10);
+    AnalysisDataService::Instance().addOrReplace(name, ws);
+    return ws;
+  }
+
+  std::vector<std::string> createWorkspaces(std::vector<std::string> const &workspaceNames = {"ws_1"}) {
+    for (auto name : workspaceNames) {
+      createWorkspace(name);
+    }
+    return workspaceNames;
+  }
+
+  const FileFormatOptions createFileFormatOptions(NamedFormat format) {
+    return FileFormatOptions(
+        /*format=*/format,
+        /*prefix=*/m_prefix,
+        /*includeHeader=*/m_includeHeader,
+        /*separator=*/m_separator,
+        /*includeQResolution=*/m_includeQResolution);
```

Comment:
What is the purpose of these comments? The variable names would appear to be enough

Reply:
I think I just copied this across from the main code and then kept it as it was. While I'm making other changes, I'll take those out.